webboot/webboot.go

```diff
@@ -23,9 +40,88 @@ var (
 	ipv6    = flag.Bool("ipv6", true, "use IPV6")
 )
 
+func wgetArgs(argURL string) (read io.ReadCloser, outFile string, werr error) {
+	//checks if argURL is a key in osSystems map
+	if _, ok := osSystems[argURL]; !ok {
+		//if argURL is not the map, use it as a URL link for wget
+		read, werr = wget(argURL)
+	} else {
+		// if argURL is in osSystems map, use its download link for wget
+		read, werr = wget(osSystems[argURL].DownloadLink)
+		outFile = argURL
+	}
+	return read, outFile, werr
+}
+
+func wget(arg string) (io.ReadCloser, error) {
+	resp, err := http.Get(arg)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("non-200 HTTP status: %d", resp.StatusCode)
+	}
+	return resp.Body, nil
+}
+
+func writer(read io.ReadCloser, fileName string) (string, error) {
+	w, err := os.Create(fileName)
+	if err != nil {
+		return "", err
+	}
+
+	defer w.Close()
+
+	_, err = io.Copy(w, read)
+	return fileName, err
+}
+
+func outputName(argURL string) (outFile string) {
```

Comment:
just call it URL, not argURL
also don't call it outthis and outthat, that' fine for microsoft C but not Go.
also it's not a File, it's a name, and you don't need to name return values in most cases, so you can just say
func outputName(URL string) (string) {

Reply:
You are right. I will keep the variable and function names simple as can be.